modules/exploits/linux/samba/is_known_pipename.rb

```diff
@@ -39,58 +38,85 @@ def initialize(info = {})
-          #
-          # Not ready yet
-          # [ 'Linux ARM (LE)',   { 'Arch' => ARCH_ARMLE } ],
-          # [ 'Linux MIPS',       { 'Arch' => MIPS } ],
+          [ 'Linux ARM (LE)',   { 'Arch' => ARCH_ARMLE } ],
+          [ 'Linux ARM64',      { 'Arch' => ARCH_AARCH64 } ],
+          [ 'Linux MIPS',       { 'Arch' => ARCH_MIPS } ],
+          [ 'Linux MIPSLE',     { 'Arch' => ARCH_MIPSLE } ],
+          [ 'Linux MIPS64',     { 'Arch' => ARCH_MIPS64 } ],
+          [ 'Linux PPC',        { 'Arch' => ARCH_PPC } ],
+          [ 'Linux PPC64',      { 'Arch' => ARCH_PPC64 } ],
+          [ 'Linux PPC64 (LE)', { 'Arch' => ARCH_PPC64LE } ],
+          [ 'Linux SPARC',      { 'Arch' => ARCH_SPARC } ],
         ],
+      'DefaultOptions' =>
+        {
+          'DCERPC::fake_bind_multi' => false,
+          'SHELL'                   => '/bin/sh',
+        },
       'Privileged'      => true,
       'DisclosureDate'  => 'Mar 24 2017',
-      'DefaultTarget'   => 1))
+      'DefaultTarget'   => 0))
 
     register_options(
       [
         OptString.new('SMB_SHARE_NAME', [false, 'The name of the SMB share containing a writeable directory']),
-        OptString.new('SMB_SHARE_BASE', [false, 'The remote filesystem path correlating with the SMB share name']),
         OptString.new('SMB_FOLDER', [false, 'The directory to use within the writeable SMB share']),
       ])
-
-    register_advanced_options(
-      [
-        OptBool.new('BruteforcePID', [false, 'Attempt to use two connections to bruteforce the PID working directory', false]),
-      ])
   end
 
-
-  def generate_common_locations
-    candidates = []
-    if datastore['SMB_SHARE_BASE'].to_s.length > 0
-      candidates << datastore['SMB_SHARE_BASE']
-    end
-
-    %W{ /volume1 /volume2 /volume3 /volume4
-        /shared /mnt /mnt/usb /media /mnt/media
-        /var/samba /tmp /home /home/shared
-    }.each do |base_name|
-      candidates << base_name
-      candidates << [base_name, @share]
-      candidates << [base_name, @share.downcase]
-      candidates << [base_name, @share.upcase]
-      candidates << [base_name, @share.capitalize]
-      candidates << [base_name, @share.gsub(" ", "_")]
-    end
-
-    candidates.uniq
+  # Setup our mapping of Metasploit architectures to gcc architectures
+  def setup
+    super
+    @@payload_arch_mappings = {
+        ARCH_X86     => [ 'x86' ],
+        ARCH_X64     => [ 'x86_64' ],
+        ARCH_MIPS    => [ 'mips' ],
+        ARCH_MIPSLE  => [ 'mipsel' ],
+        ARCH_MIPSBE  => [ 'mips' ],
+        ARCH_MIPS64  => [ 'mips64' ],
+        ARCH_PPC     => [ 'powerpc' ],
+        ARCH_PPC64   => [ 'powerpc64' ],
+        ARCH_PPC64LE => [ 'powerpc64le' ],
+        ARCH_SPARC   => [ 'sparc' ],
+        ARCH_ARMLE   => [ 'armel', 'armhf' ],
+        ARCH_AARCH64 => [ 'aarch64' ],
+        #
+        # Missing from Rex::Arch
+        #   ARCH_MIPS64LE = 'mips64el',
+        #   ARCH_SPARC64  = 'sparc64',
+        #   ARCH_S390X    = 's390x',
```

Comment:
Added the missing architectures here: https://github.com/rapid7/rex-arch/pull/3

Will land and update this PR to match.